Python/codegen.c

```diff
@@ -1826,11 +1826,13 @@ codegen_jump_if(compiler *c, location loc,
                 ADDOP_COMPARE(c, LOC(e), asdl_seq_GET(e->v.Compare.ops, i));
                 ADDOP(c, LOC(e), TO_BOOL);
                 ADDOP_JUMP(c, LOC(e), POP_JUMP_IF_FALSE, cleanup);
+                ADDOP(c, NO_LOCATION, NOT_TAKEN);
             }
             VISIT(c, expr, (expr_ty)asdl_seq_GET(e->v.Compare.comparators, n));
             ADDOP_COMPARE(c, LOC(e), asdl_seq_GET(e->v.Compare.ops, n));
             ADDOP(c, LOC(e), TO_BOOL);
             ADDOP_JUMP(c, LOC(e), cond ? POP_JUMP_IF_TRUE : POP_JUMP_IF_FALSE, next);
+            ADDOP(c, NO_LOCATION, NOT_TAKEN);
```

Comment:
The peephole optimiser replaces a conditional jump with const condition by an unconditional jump and a NOP. We probably need another peepholer rule to remove any NOT_TAKEN this leaves behind.

Reply:
I worry this might be not work if the optimizer merges are rearranges branches.
It is possible that the correspondence to the original source could be lost.